lib/node_modules/@stdlib/math/base/special/ellipj/lib/assign.js

```diff
@@ -0,0 +1,237 @@
+*
+* @private
+* @param {number} u - input value
+* @param {number} m - modulus `m`, equivalent to `k²`
+* @param {Collection} out - output array
+* @param {integer} stride - output array stride
+* @param {NonNegativeInteger} offset - output array index offset
+* @throws {Error} Throws error in the event of internal overflow.
+* @returns {Collection} output array
+*
+* @example
+* var ellipj = require( '@stdlib/math/base/special/ellipj' ).assign;
+* var v = ellipj( 0.3, 0.5, [ 0.0, 0.0, 0.0, 0.0 ], 1, 0 );
+* // returns [ ~0.293, ~0.956, ~0.978, ~0.298 ]
+*
+* @example
+* v = ellipj( 0.0, 0.0, [ 0.0, 0.0, 0.0, 0.0 ], 1, 0 );
+* // returns [ ~0.0, ~1.0, ~1.0, ~0.0 ]
+*
+* @example
+* v = ellipj( Infinity, 1.0, [ 0.0, 0.0, 0.0, 0.0 ], 1, 0 );
+* // returns [ ~1.0, ~0.0, ~0.0, ~1.571 ]
+*
+* @example
+* v = ellipj( 0.0, -2.0, [ 0.0, 0.0, 0.0, 0.0 ], 1, 0 );
+* // returns [ ~0.0, ~1.0, ~1.0, NaN ]
+*
+* @example
+* v = ellipj( NaN, NaN, [ 0.0, 0.0, 0.0, 0.0 ], 1, 0 );
+* // returns [ NaN, NaN, NaN, NaN ]
+*/
+function assign(u, m, out, stride, offset) {
+	var dnDenom;
+	var uK2cen;
+	var k1inv;
+	var sechu;
+	var sinhu;
+	var tanhu;
+	var phi0;
+	var phi1;
+	var atmp;
+	var FLG;
+	var gdu;
+	var uK2;
+	var uK4;
+	var mu;
+	var K2;
+	var K4;
+	var u0;
+	var sn;
+	var cn;
+	var dn;
+	var am;
+	var k;
+	var s;
+	var c;
+	var f;
+	var a;
+	var b;
+	var N;
+
+	if ( m < 0.0 ) {
+		// A&S 16.10.1 for a negative parameter, mapping -m to 0 < mu < 1
+		mu = -m / ( 1.0 - m );
+		k1inv = sqrt( 1.0 - m );
+		assign( u * k1inv, mu, tmp4, 1, 0 );
+		sn = ( tmp4[ 0 ] / tmp4[ 2 ] ) / k1inv;
+		cn = tmp4[ 1 ] / tmp4[ 2 ];
+		dn = 1.0 / tmp4[ 2 ];
+		am = NaN;
+	} else if ( m > 1.0 ) {
+		// A&S 16.11.1 for reciprocal parameter, mapping m > 1 to 0 < mu < 1
+		k = sqrt( m );
+		assign( u * k, 1.0 / m, tmp4, 1, 0 );
+		sn = tmp4[ 0 ] / k;
+		cn = tmp4[ 2 ];
+		dn = tmp4[ 1 ];
+		am = NaN;
+	} else if ( m === 0.0 ) {
+		// A&S table 16.6, limiting case m = 0: circular trigonometric functions
+		sincos( u, tmp2, 1, 0 );
+		sn = tmp2[ 0 ];
+		cn = tmp2[ 1 ];
+		dn = 1.0;
+		am = u;
+	} else if ( m === 1.0 ) {
+		// A&S table 16.6: limiting case m = 1: hyperbolic functions
+		sn = tanh( u );
+		cn = 1.0 / cosh( u );
+		dn = cn;
+		am = atan( sinh( u ) );
+	} else if ( m < SQRT_EPS ) {
+		// A&S 16.13.1 for small-u approximations. Additionally, compute K at extra cost in
+		// Order to ensure returned values are correct outside the range [0, K].
+		K4 = 4.0 * ellipk( m );
+		u0 = ( ( u % K4 ) + K4 ) % K4;
+		sincos( u0, tmp2, 1, 0 );
+		s = tmp2[ 0 ];
+		c = tmp2[ 1 ];
+		f = 0.25 * m * ( u0 - ( s * c ) );
+		sn = s - ( f * c );
+		cn = c + ( f * s );
+		dn = 1.0 - ( 0.5 * m * s * s );
+		am = u - ( 0.25 * m * ( u - ( s * c ) ) );
+	} else if ( m > 1.0 - SQRT_EPS ) {
+		// A&S 16.15.1 - 16.15.4 for near-unity approximations. Additionally, compute K at
+		// Extra cost so that we may reflect as needed to ensure the returned values
+		// Are correct.
```

Comment:
```suggestion
		// A&S 16.15.1 - 16.15.4 for near-unity approximations. Additionally, compute K at extra cost so that we may reflect as needed to ensure the returned values are correct.
```